src/qat/purr/backends/qblox/device.py

```diff
@@ -253,6 +253,11 @@ def start_playback(self, repetitions: int, repetition_time: float):
             raise ValueError("No resources allocated. Install packages first")
 
         results = {}
+
+        for module, allocations in self._resources.items():
```

Comment:
Is it strictly necessary to redo this iteration over and over? :thinking:

Reply:
This would look cleaner pulled out as a short helper function with a good name...

It looks like the code flow is something like ...

```
self._sync_resources(True)
*do stuff*
self._sync_resources(False)
```

Reply:
yep, it's necessary ^^

Reply:
> This would look cleaner pulled out as a short helper function with a good name...
> 
> It looks like the code flow is something like ...
> 
> ```
> self._sync_resources(True)
> *do stuff*
> self._sync_resources(False)
> ```

it's a nice but minor improvement given the scale of other things, it'll part of another PR